docs/src/05adr-test-execution.md

```diff
@@ -0,0 +1,70 @@
+# ADR-05: Test execution
+
+| authors          | revision | revision date  |
+| ---------------- | --------:| --------------:|
+| Andrey Kuprianov |        1 | July 18, 2022  |
+
+The present ADR describes the external interfaces of the `Execute` component that deals with test execution, either from a given abstract trace, or from (a set of) traces generated from a TLA+ model.
+
+
+## Command line interface
+
+Test execution CLI provides two main entry points:
+
+- `test trace [--trace <trace-file>] [--reactor <reactor-file>]` allows to execute an abstract test trace against the chain. All additional parameters are optional; when omitted, the defaults will be used.
+  - `--trace <trace-file>` allows to specify which test trace file to execute. 
+  - `--reactor <reactor-file>` allows to provide explicitly the reactor to employ; when omitted, the reactor previously generated via `reactor` command will be used.
+- `test model [--model <model-file>] [--config <config-file>] [--sample <sample-operator>]` allows to execute test traces generated from a TLA+ model. All additional parameters are optional; when omitted, the defaults will be used.
+  - `--model <model-file>` allows to specify which TLA+ model to use.
+  - `--config <config-file>` specifies the model configuration file.
+  - `--sample <sample-operator>` gives the name of the sample operator describing the test traces.
+
+**TODO**: add configuration capabilities, e.g.:
+- `test set timeout <TIMEOUT>` allows to specify the timeout (in seconds) for execution of a single test.
+
+## Programmatic dependencies
+
+`Execute` component depends on all three upstream components:
+
+### Setup
+
+From `Setup` component it needs three functions provided via an agreed upon entry point:
+- `setup()` sets up the chain with the default configured parameters;
+  - **TODO**: agree upon the way to provide initialization parameters (e.g. the set of users and their wallets)from the trace into the chain genesis.
+- `get_client()` provides the configured `LCDClient` ready to be used with the chain;
+- `teardown()` destroys the previously set up chain.
+
+## Model
+
+From `Model` component it needs two functions provided via an agreed upon entry point:
+- `get_trace(trace = None)` retrieves the ITF trace, either the default one, or from the provided location;
+- `get_model_trace(model = None, config = None, sample = None)` provides a trace from the model, using either default, or the provided parameters.
+
+## React (Generate)
```

Comment:
`React` calls different associations than Reactor (I would guess most people would connect it somehow to reactive programming. I suggest the name `ReactorRoom` (as in my PR) or to settle on plain `Generate`

Reply:
In fact, I like the name `Reactor`. Short and clear enough. I just caught myself at calling your component that way in my mind:)